### PR TITLE
Add integration doc for AKS Dapr extension and clean up hosting k8s pages

### PR DESCRIPTION
**Pull request by greenie-msft**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

In the Kubernetes cluster creation doc page, added instructions on how to install Dapr via the AKS Dapr extension

## Issue reference

https://github.com/dapr/docs/issues/2102

### CODE REVIEW

**Comment by msfussell** [comment]
Feedback on how Dapr is deployed to K8s clusters

1) In the kubernetes-deploy.md doc the Create cluster section (see copied below) is inconsistent. It should 
a) include a link to the KiND cluster that has been added 
b) The links should be to the GKE doc here https://docs.dapr.io/operations/hosting/kubernetes/cluster/setup-gke/ with this GKE article being updated to have the link to the GKE quickstart, this one https://cloud.google.com/kubernetes-engine/docs/quickstart

Create cluster
You can install Dapr on any Kubernetes cluster. Here are some helpful links:

[Setup Minikube Cluster]({{< ref setup-minikube.md >}})
[Setup Azure Kubernetes Service Cluster]({{< ref setup-aks.md >}})
Setup Google Cloud Kubernetes Engine
Setup Amazon Elastic Kubernetes Service

2) Can re reorder the Index here https://docs.dapr.io/operations/hosting/kubernetes/ so that is goes
- Overview of Dapr on Kubernetes
- Kubernetes cluster setup
- Deploy Dapr on a Kubernetes cluster
in that order

**Comment by greenie-msft** [comment]
@msfussell can you clarify where this should be added 

>  with this GKE article being updated to have the link to the GKE quickstart, this one https://cloud.google.com/kubernetes-engine/docs/quickstart

File: daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+If the `k8s-extension` extension is already present, you can update it to the latest version using the below command:
+
+```bash
+az extension update --name k8s-extension
+```
+
+#### Create the extension and install Dapr on your AKS cluster
+Once your subscription is registered to use Kubernetes extensions, you can create the Dapr extension, which installs Dapr on your AKS cluster. For example:
```
Comment (Nickomang) [inline daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md]: "After your subscription is registered to use Kubernetes extensions, install Dapr on your cluster by creating the Dapr extension. For example:"

File: daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+```bash
+az k8s-extension create --cluster-type managedClusters \
+--cluster-name myAKSCluster \
+--resource-group myResourceGroup \
+--name myDaprExtension \
+--extension-type Microsoft.Dapr
+```
+
+Additionally, you also have the option of allowing Dapr to auto-update its minor version by specifying the `--auto-upgrade-minor-version` parameter and setting the value to true:
```
Comment (Nickomang) [inline daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md]: "Additionally, Dapr can automatically update its minor version. To enable this, set the `--auto-upgrade-minor-version` parameter to true:"

File: daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+{{% alert title="Note" color="warning" %}}
+If you install Dapr through the AKS extension, our recommendation is to continue using the extension for future management of Dapr instead of the Dapr CLI. Combining the two tools can cause conflicts and result in undesired behavior.
+{{% /alert %}}
+
+### How the extension works
+The Dapr extension uses the Azure CLI to provision the Dapr control plane on your AKS cluster. The Dapr control plane consists of:
+
+- **dapr-operator**: Manages component updates and Kubernetes services endpoints for Dapr (state stores, pub/subs, etc.)
+- **dapr-sidecar-injector**: Injects Dapr into annotated deployment pods and adds the environment variables DAPR_HTTP_PORT and DAPR_GRPC_PORT to enable user-defined applications to easily communicate with Dapr without hard-coding Dapr port values.
```
Comment (Nickomang) [inline daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md]: "Injects Dapr into annotated deployment pods and adds the environment variables DAPR_HTTP_PORT and DAPR_GRPC_PORT. This enables user-defined applications to communicate with Dapr without the need to hard-code Dapr port values."

File: daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## Install Dapr using the AKS Dapr extension
+You have the option of using the AKS Dapr extension to provision Dapr on your AKS cluster. Additionally, the extension offers support for all native Dapr configuration capabilities through command-line arguments via the Azure CLI and offers the option of opting into automatic minor version upgrades of the Dapr runtime.
+
+{{% alert title="Note" color="warning" %}}
+If you install Dapr through the AKS extension, our recommendation is to continue using the extension for future management of Dapr instead of the Dapr CLI. Combining the two tools can cause conflicts and result in undesired behavior.
+{{% /alert %}}
+
+### How the extension works
+The Dapr extension uses the Azure CLI to provision the Dapr control plane on your AKS cluster. The Dapr control plane consists of:
```
Comment (Nickomang) [inline daprdocs/content/en/developing-applications/integrations/Azure/azure-kubernetes-service-extension.md]: "The Dapr extension works by provisioning the Dapr control plane on your AKS cluster through the Azure CLI. The dapr control plane consists of:"

**Review by Nickomang — APPROVED** [review]

**Comment by greenie-msft** [comment]
Closing this PR since merge conflicts were complex. New PR with feedback incorporated targeting v1.6 base branch here: https://github.com/dapr/docs/pull/2103

**Pull request closed without merging**